specification/Metadata/README.md

```diff
@@ -0,0 +1,990 @@
+
+JSON encoding is useful for encoding data where readability matters. This works well for small amounts of data, but does not scale well to large datasets. If the metadata is expected to grow large, binary encoding would be a better choice. One situation where JSON encoding is helpful is if metadata will be edited by hand, as JSON is easier for a human to understand than editing a binary buffer.
+
+Binary encoding is designed for storage efficiency, and is designed for use with large datasets. Data is packed in parallel arrays, one per property. This allows for storage optimizations based on data type, such as storing boolean properties as a tightly packed bit vector. This encoding is more involved than the JSON encoding, but it is much preferred in most cases where performance is an important consideration.
+
+These first two encodings are designed for discrete properties indexed by instance ID. In contrast, metadata textures are used when instances are identified by spatial position (i.e. texture coordinates within a texture). Heightmaps and normal maps are two examples. This type of per-texel metadata has many uses, but is also somewhat limited by the image formats used to store data.
+
+Comparison table: 
+
+|   | JSON | Binary | Texture |
+|---|------|--------|---------|
+| Supports large datasets | No | Yes | Yes |
+| Supports small datasets | Yes | Yes | Yes |
+| Is human-readable | Yes | No | No |
+| Indexed by instance IDs | Yes | Yes | No |
+| Indexed by texture coordinates | No | No | Yes |
+| Supports Optional Values | Yes | No | No |
+
+
+### Separate Class Definition from Instantiation
+
+This specification keeps class definition separate from instantiation. While this adds a level of indirection, there are several reasons for this separation:
+
+* It allows the class definition to be stored seperately from the data itself. This is useful when one has a dataset that has many 3D models with the same shape of metadata. Then instead of re-declaring the class many times, a single definition can be shared across multiple models
+* It allows greater flexibility for storing properties. For example, take the `elevation` property from the example above. The natural representation is to use a metadata texture. However, if the corresponding 3D model is a high-resolution terrain mesh, storing the elevations per-vertex may be more appropriate.
+
+## Class Definitions
+
+### Basic Types
+
+A class is a collection of properties. Each class requires a unique class ID. This can be any UTF-8 string. The uniqueness is enforced by a JSON dictionary. Within the class, one or more properties is defined. Each property has a type associated with it. Some types have extra options; the sections below discuss such cases.
+
+Classes are defined in a `classes` section of the JSON file 
```

Comment:
I'd remove this. It's on the implementing spec to specify where `classes` are declared.